homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -27,10 +27,26 @@ async def async_step_user(self, user_input=None) -> FlowResult:
         if user_input:
             address = user_input[CONF_ADDRESS]
 
-            if await self._async_is_server_online(address):
-                # No error was detected, create configuration entry.
-                config_data = {CONF_NAME: user_input[CONF_NAME], CONF_ADDRESS: address}
-                return self.async_create_entry(title=address, data=config_data)
+            # Prepare config entry data.
+            config_data = {
+                CONF_NAME: user_input[CONF_NAME],
+                CONF_ADDRESS: address,
+            }
+
+            # Some Bedrock Edition servers mimic a Java Edition server, therefore check for a Bedrock Edition server first.
+            for server_type in MinecraftServerType:
+                try:
+                    server = MinecraftServer(server_type, address)
```

Comment:
Please run the creation of the server in the executor as the lookup is performed too